src/rattler_build_conda_compat/modify_recipe.py

```diff
@@ -0,0 +1,142 @@
+from __future__ import annotations
+
+import hashlib
+import io
+import re
+from typing import TYPE_CHECKING, Any, Generator
+
+import requests
+from ruamel.yaml import YAML
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+yaml = YAML()
+yaml.preserve_quotes = True
+
+
+def _update_build_number_in_context(recipe: dict[str, Any], new_build_number: int) -> bool:
+    for key in recipe.get("context", {}):
+        if key.startswith("build_") or key == "build":
+            recipe["context"][key] = new_build_number
+            return True
+    return False
+
+
+def _update_build_number_in_recipe(recipe: dict[str, Any], new_build_number: int) -> bool:
+    is_modified = False
+    if "build" in recipe and "number" in recipe["build"]:
+        recipe["build"]["number"] = new_build_number
+        is_modified = True
+
+    if "outputs" in recipe:
+        for output in recipe["outputs"]:
+            if "build" in output and "number" in output["build"]:
+                output["build"]["number"] = new_build_number
+                is_modified = True
+
+    return is_modified
+
+
+def update_build_number(file: Path, new_build_number: int) -> str:
+    # This function should be called to update the build number of the recipe
+    # in the meta.yaml file.
+    with file.open("r") as f:
+        data = yaml.load(f)
+    build_number_modified = _update_build_number_in_context(data, new_build_number)
+    if not build_number_modified:
+        build_number_modified = _update_build_number_in_recipe(data, new_build_number)
+
+    with io.StringIO() as f:
+        yaml.dump(data, f)
+        return f.getvalue()
+
+
+class CouldNotUpdateVersionError(Exception):
+    NO_CONTEXT = "Could not find context in recipe"
+    NO_VERSION = "Could not find version in recipe context"
+
+    def __init__(self, message: str = "Could not update version") -> None:
+        self.message = message
+        super().__init__(self.message)
+
+
+class Hash:
+    def __init__(self, hash_type: str, hash_value: str) -> None:
+        self.hash_type = hash_type
+        self.hash_value = hash_value
+
+    def __str__(self) -> str:
+        return f"{self.hash_type}: {self.hash_value}"
+
+
+def has_jinja_version(url: str) -> bool:
```

Comment:
I think this also can be moved in utils.py as it seems something useful potentially to have for other modules